R/jointRPCA.R

```diff
@@ -0,0 +1,752 @@
+        names(tables) <- if (is.null(nm) || is.na(nm)) "assay1" else nm
+        
+    } else if (is.list(x) && all(vapply(x, is.matrix, logical(1)))) {
+        tables <- x
+        if (is.null(names(tables))) {
+            names(tables) <- paste0("view", seq_along(tables))
+        }
+        
+    } else if (is.matrix(x)) {
+        tables <- list(x)
+        names(tables) <- "assay1"
+        
+    } else {
+        stop(
+            "Unsupported input type for jointRPCAuniversal(): ",
+            paste(class(x), collapse = ", ")
+        )
+    }
+    
+    res <- .jointRPCA(tables = tables, ...)
+    
+    if (inherits(x, "MultiAssayExperiment")) {
+        res$experiment.names <- experiments
+        res$assay.names.used <- assay_names_used
+    }
+    
+    res
+}
+
+#' Run Joint-RPCA and store embedding in reducedDim
+#'
+#' Convenience wrapper that runs Joint Robust PCA on one or more compositional
+#' tables and stores the resulting sample embedding in \code{reducedDim(x, name)},
+#' similar to \code{runMDS()} and \code{runPCA()}.
+#'
+#' @param x A \code{SummarizedExperiment}, \code{TreeSummarizedExperiment},
+#'   \code{MultiAssayExperiment}, or a related object supported by
+#'   \code{jointRPCAuniversal()}.
+#' @param experiments Optional character vector of experiment names to use when
+#'   \code{x} is a \code{MultiAssayExperiment} (i.e. \code{names(experiments(x))}).
+#'   Ignored for \code{SummarizedExperiment} inputs.
+#' @param altexp Optional name of an alternative experiment. If supplied,
+#'   Joint-RPCA is run on \code{altExp(x, altexp)} instead of \code{x}.
+#' @param name Character scalar giving the name of the \code{reducedDim} slot
+#'   in which to store the joint sample embedding. Defaults to \code{"JointRPCA"}.
+#' @param ... Additional arguments passed to \code{jointRPCAuniversal()},
+#'   and from there to the internal \code{.jointRPCA()} engine (e.g. \code{n.components},
+#'   \code{min.sample.count}, \code{max.iterations}, etc.).
+#'
+#' @return The input object \code{x} with a new entry in
+#'   \code{reducedDim(x, name)} containing the Joint-RPCA sample embedding.
+#'   The full Joint-RPCA result (including distances, cross-validation
+#'   statistics and transformed tables) is stored in
+#'   \code{metadata(x)$JointRPCA[[name]]}.
+#'
+#' @export
+
+runJointRPCA <- function(x,
+                         experiments = NULL,
+                         altexp = NULL,
+                         name   = "JointRPCA",
+                         ...) {
+    #select the object to operate on
+    y <- x
+    if (!is.null(altexp)) {
+        y <- altExp(x, altexp)
+    }
+    
+    #use universal front-end to build tables + run .jointRPCA()
+    res <- jointRPCAuniversal(y, experiments = experiments, ...)
+    
+    #extract sample embedding
+    emb <- res$ord.res$samples
+    if (is.null(emb) || !is.matrix(emb)) {
+        stop("Joint-RPCA did not return a valid sample embedding in ord.res$samples.")
+    }
```

Comment:
User cannot solve this error message. We should make fix the function that it does not lead to this error.